web/src/main/java/org/wso2/testgrid/web/api/SSOService.java

```diff
@@ -81,9 +83,14 @@ public Response createSession(@FormParam("SAMLResponse") String responseMessage)
         try {
             validateSignature(saml2Response.getSignature());
             request.getSession();
+            String redirectUri;
+            if (StringUtil.isStringNullOrEmpty(relayState)) {
+                redirectUri = Constants.WEBAPP_CONTEXT;
+            } else {
+                redirectUri = relayState;
+            }
```

Comment:
In summary, this is actually:
redirectUri = StringUtil.isStringNullOrEmpty(relayState) ? Constants.WEBAPP_CONTEXT : relayState;
:)